src/test/compile-fail/lifetime-inference-give-expl-lifetime-param.rs

```diff
@@ -39,8 +39,7 @@ struct Cat<'x, T> { cat: &'x isize, t: T }
 struct Dog<'y> { dog: &'y isize }
 
 fn cat2<'x, 'y>(x: Cat<'x, Dog<'y>>) -> &'x isize {
-//~^ HELP: consider using an explicit lifetime parameter as shown: fn cat2<'x>(x: Cat<'x, Dog<'x>>) -> &'x isize
-    x.t.dog //~ ERROR: cannot infer
+    x.t.dog //~ ERROR E0312
```

Comment:
I was debating whether to consinder this a "regression". I've actually been considering removing the code for those messages, since it is out of date now that we have `:` where-clauses, and I've not found it helpful of late. But I don't know if it is helpful for beginners or not. 

The main place I DO find it handy sometimes though is telling me when some struct type has implicit regions I forgot about.